hypervisor/arch/x86/vtd.c

```diff
@@ -85,18 +85,18 @@ dmar_set_bitslice(uint64_t var, uint64_t mask,
 
 #define DMAR_OP_TIMEOUT CYCLES_PER_MS
 
-#define DMAR_WAIT_COMPLETION(offset, condition, status) \
-	do {                                                \
+#define DMAR_WAIT_COMPLETION(offset, condition, status)			\
```

Comment:
This is also API like macro, can we change it to inline API?

Reply:
'status' is used to calculate the value of 'condition' when we use DMAR_WAIT_COMPLETION. And how to calculate 'condition' is case by case. We could not convert it to a API.

Reply:
we can change the semantics of MACO and/or code logic for this.